server/iframe.go

```diff
@@ -222,14 +261,98 @@ func (a *API) authenticate(w http.ResponseWriter, r *http.Request) {
 	http.SetCookie(w, csrfCookie)
 
 	// Redirect to the home page
-	http.Redirect(w, r, "/", http.StatusSeeOther)
+	http.Redirect(w, r, redirectPath, http.StatusSeeOther)
+}
+
+// MessageHasBeenPosted is called when a message is posted in Mattermost. We rely on it to send a user activity notification
+// to Microsoft Teams when a user is mentioned in a message.
+// This is called in a controller Goroutine in the server side so there's no need to worry about concurrency here.
+func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
```

Comment:
`MessageHasBeenPosted` tends to be expensive -- can we preflight looking up the user's MS Teams ID as early as possible and avoid doing any other work if we can abort early?

Reply:
I think we already do it as early as possible, right? We extract the mentions and check each user for the properly.

Reply:
The author of the post may not have O365 id stored, but we still need to send notifications for the users mentioned.

Reply:
I think I was just looking at the log, and `context`/`jsonContext`/`urlParams` initialization -- admittedly none of that really exercises the pluginAPI (besides the log) so it's not a big deal.

But I'm curious, could we use the `NotificationWillBePushed` hook instead?

(Unblocking my requested changes as we can iterate on this.)